sstables/sstable_mutation_reader.hh

```diff
@@ -6,7 +6,7 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 #include "mutation/mutation.hh"
-#include "types.hh"
+#include "types/types.hh"
```

Comment:
Ditto, same question on which types.hh was really intended here, and maybe you're right and what we had here was actually wrong.

I see the same problem (or non-problem) in a few other files in sstables/ below, so please take a look at all of them.